code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -177,6 +177,8 @@
 
 /obj/structure/closet/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
 	if(istype(W, /obj/item/weapon/rcs) && !src.opened)
+		if(locate(user) in src.contents) //to prevent self-teleporting.

```

Comment:
.. this is the most terrifying way you could possibly do this. 
If anything, it should look more like
```
if(user in contents) return
```